Example/src/main/java/com/infomaniak/login/example/LoginActivity.kt

```diff
@@ -31,35 +33,46 @@ class LoginActivity : AppCompatActivity() {
                 lifecycleScope.launch {
                     val okHttpClient = OkHttpClient.Builder().build()
 
-                    infomaniakLogin.getToken(okHttpClient, it,
-                        { apiToken ->
-                            getTokenStatus.text = "${apiToken.userId} ${apiToken.accessToken}"
-
-                            btnDeconnect.apply {
-                                isVisible = true
-                                setOnClickListener { infomaniakLogin.deconnect(okHttpClient, apiToken) }
-                            }
-                        }, { error ->
-                            when (error) {
-                                InfomaniakLogin.ErrorStatus.SERVER -> getTokenStatus.text = "Server error"
-                                else -> getTokenStatus.text = "Error"
-                            }
-                        }
-                    )
+                    val tokenResult = infomaniakLogin.getToken(okHttpClient, code = it)
+                    Dispatchers.Main {
+                        updateUi(tokenResult, infomaniakLogin, okHttpClient)
+                    }
                 }
             }
         }
     }
 
-    private fun InfomaniakLogin.deconnect(okHttpClient: OkHttpClient, token: ApiToken) = with(binding) {
+    private fun ActivityLoginBinding.updateUi(
+        tokenResult: InfomaniakLogin.TokenResult,
+        infomaniakLogin: InfomaniakLogin,
+        okHttpClient: OkHttpClient,
+    ) {
+        if (tokenResult is InfomaniakLogin.TokenResult.Success) {
+            val apiToken = tokenResult.apiToken
+            getTokenStatus.text = "${apiToken.userId} ${apiToken.accessToken}"
+
+            btnDeconnect.apply {
+                isVisible = true
+                setOnClickListener { infomaniakLogin.logout(okHttpClient, apiToken) }
+            }
+        } else {
+            when ((tokenResult as InfomaniakLogin.TokenResult.Error).errorStatus) {
+                InfomaniakLogin.ErrorStatus.SERVER -> getTokenStatus.text = "Server error"
+                else -> getTokenStatus.text = "Error"
+            }
+        }
+    }
```

Comment:
IMHO, this would be a clearer replacing the if/else with a when with subject, and it would avoid this explicit cast: `okenResult as InfomaniakLogin.TokenResult.Error`